src/js/live-tracker.js

```diff
@@ -0,0 +1,204 @@
+    // and we want some wiggle room for short segment live playback
+    const liveEdgeWindow = seekableIncrement + 0.07;
+
+    return liveCurrentTime !== Infinity && (liveCurrentTime - liveEdgeWindow) >= currentTime;
+  }
+
+  // all the functionality for tracking when seek end changes
+  // and for tracking how far past seek end we should be
+  trackLive_() {
+    this.pastSeekEnd_ = this.pastSeekEnd_ || 0;
+    const seekable = this.player_.seekable();
+
+    // skip undefined seekable
+    if (!seekable || !seekable.length) {
+      return;
+    }
+
+    const newSeekEnd = this.seekableEnd(0);
+
+    // we can only tell if we are behind live, when seekable changes
+    // once we detect that seekable has changed we check the new seek
+    // end against current time, with a fudge value of half a second.
+    if (newSeekEnd !== this.lastSeekEnd_) {
+      if (this.lastSeekEnd_) {
+        this.seekableIncrement_ = newSeekEnd - this.lastSeekEnd_;
+      }
+
+      this.pastSeekEnd_ = 0;
+      this.lastSeekEnd_ = newSeekEnd;
+      this.trigger('seekableendchange');
+    }
+
+    this.pastSeekEnd_ = this.pastSeekEnd() + 0.03;
+
+    if (this.isBehind_() !== this.behindLiveEdge()) {
+      this.behindLiveEdge_ = this.isBehind_();
+      this.trigger('liveedgechange');
+    }
+  }
+
+  /**
+   * handle a durationchange event on the player
+   * and start/stop tracking accordingly.
+   */
+  handleDurationchange() {
+    if (this.player_.duration() === Infinity) {
+      this.start();
+    } else {
+      this.stop();
+    }
+  }
+
+  /**
+   * start tracking live playback
+   */
+  start() {
+    if (this.started()) {
+      return;
+    }
+
+    this.trackingInterval_ = this.setInterval(this.trackLive_, 30);
+    this.trackLive_();
+  }
+
+  /**
+   * stop tracking live playback
+   */
+  stop() {
+    if (!this.started()) {
+      return;
+    }
+    this.pastSeekEnd_ = 0;
+    this.lastSeekEnd_ = null;
+    this.behindLiveEdge_ = null;
+
+    this.clearInterval(this.trackingInterval_);
+    this.trackingInterval_ = null;
+  }
+
+  /**
+   * A helper to get the player seekable end
+   * so that we don't have to null check everywhere
+   */
+  seekableEnd() {
+    const seekable = this.player_.seekable();
+
+    if (!seekable || !seekable.length) {
+      return Infinity;
+    }
+
+    return seekable.end(0);
+  }
+
+  /**
+   * A helper to get the player seekable start
+   * so that we don't have to null check everywhere
+   */
+  seekableStart() {
+    const seekable = this.player_.seekable();
+
+    if (!seekable || !seekable.length) {
+      return 0;
+    }
+
+    return seekable.start(0);
+  }
+
+  /**
+   * Get the live time window
+   */
+  liveWindow() {
+    return this.seekableEnd() - this.seekableStart();
+  }
+
+  /**
+   * Determines if the player is live, only checks if this component
+   * is tracking live playback or not
+   */
+  isLive() {
+    return this.started();
```

Comment:
if someone calls `liveTracker.stop()` this will start reporting false for `isLive` even if we are live. Maybe we should continue verifying that `duration` from the video element is `Infinity`?

Reply:
The problem is that since we are not tracking live when it is stopped. A lot of the UI won't work correctly if the liveTracker is stopped during live playback. I think it might be better to revert to liveDisplay if live-tracker is stopped during live playback.